test/test_sanitizer_log_parser.py

```diff
@@ -124,3 +138,30 @@ def test_csv_sample_stack_trace_lines_look_like_stack_trace(
     for line in sanitizer_log_parser_fixture.report_csv:
         for sample_stack_trace_line in line['sample_stack_trace'].split('\n'):
             assert sample_stack_trace_line.lstrip().startswith('#')
+
+
+def test_xml_has_expected_testcases(sanitizer_log_parser_fixture) -> None:
+    assert sanitizer_log_parser_fixture.report_xml.get('tests') == \
+        sanitizer_log_parser_fixture.expected_xml.get('tests')
+    assert len(sanitizer_log_parser_fixture.report_xml.findall('testcase')) == \
+        len(sanitizer_log_parser_fixture.expected_xml.findall('testcase'))
+
+
+def test_xml_has_expected_testcase_attributes(sanitizer_log_parser_fixture) -> None:
+    for case in sanitizer_log_parser_fixture.report_xml.findall('testcase'):
+        if case is not None:
+            assert len(case.get('name')) > 0
+            assert len(case.get('errors')) > 0
+
+        for error in case.findall('error'):
```

Comment:
Seeing case.findall here makes me think that case can never be None. Why do the check on line 152?

Reply:
Good point. `case` can be None if there are no testcases executed (like in the `no_error/` scenario here). In that case, it doesn't make sense to check if the testcase reported any errors or not. Modified in the next version.